_examples/json/json/parser.go

```diff
@@ -0,0 +1,55 @@
+// Code generated by antlr2participle. DO NOT EDIT.
+package json
+
+import (
+	"github.com/alecthomas/participle/v2"
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+var (
+	Rules = lexer.Rules{
+		"Root": {
+			{"STRING", `"(\\(["\\/bfnrt]|u[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F])|[^"\\\x{0000}-\x{001F}])*"`, nil},
+			{"NUMBER", `-?(0|[1-9][0-9]*)(\.[0-9]+)?([Ee][+\-]?(0|[1-9][0-9]*))?`, nil},
+			{"ws", `[ \t\n\r]+`, nil},
+			{"XXX__LITERAL_Comma", `,`, nil},
```

Comment:
Tricky because that can change the behavior of the lexer.  By default, Go's regex engine does not use leftmost-longest.  Just ran into this in the Cap'nProto generated parser, as an example.  But I did make it an option and it's default-on so we assume the best case and if you need to explode the literals to get it to work, you can.

Reply:
> Go's regex engine does not use leftmost-longest

It's not clear to me why this is an issue. If you order the literals by longest to shortest there shouldn't be any issues, eg. `!=|!` will work as expected.

Reply:
If this is the default now can we regenerate these examples to use it?

Reply:
> > Go's regex engine does not use leftmost-longest
> 
> It's not clear to me why this is an issue. If you order the literals by longest to shortest there shouldn't be any issues, eg. `!=|!` will work as expected.

Ah, didn't think of doing that.  Silly me.  Since I already had sorting when the literals were exploded into their own rules, I solved it by retaining the ability to explode them.  I suppose I could drop that and switch to sorting the literals within the regex, but I'm not keen on investing that effort at the moment.  Let me know if you feel strongly about this.

> If this is the default now can we regenerate these examples to use it?

Most of the examples are this way now, see [the json parser](https://github.com/tooolbox/participle/blob/antlr4/_examples/json/json/parser.go#L15) for example.  I think I only had to keep [Cap'n Proto](https://github.com/tooolbox/participle/blob/antlr4/_examples/capnproto/capnproto/parser.go) exploded because of ambiguities.  Maybe `-` vs. `->` but I don't remember, all I recall is the tests started failing and I reverted it.